app/src/main/kotlin/com/wire/android/ui/WireActivity.kt

```diff
@@ -1,25 +1,115 @@
+import androidx.compose.material3.DrawerState
+import androidx.compose.material3.DrawerValue
 import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.NavigationDrawer
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.rememberDrawerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.rememberNavController
+import com.wire.android.navigation.*
+import com.wire.android.ui.drawer.WireDrawer
 import com.wire.android.ui.theme.WireTheme
+import com.wire.android.ui.topbar.WireTopBar
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
 @ExperimentalMaterial3Api
 @ExperimentalAnimationApi
 @ExperimentalComposeUiApi
-class WireActivity : ComponentActivity() {
+@AndroidEntryPoint
+class WireActivity : AppCompatActivity() {
+
+    @Inject
+    lateinit var navigationManager: NavigationManager
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             WireTheme {
-                WireApp()
+
+                val navController = rememberNavController()
+                val drawerState = rememberDrawerState(DrawerValue.Closed)
+                val currentItem = navController.getCurrentNavigationItem()
+                val scope = rememberCoroutineScope()
+
+                setUpNavigation(drawerState, navController, scope)
+
+                val topBar = WireTopBar(navigationElements = currentItem?.navigationElements)
+                val drawerContent = WireDrawer(currentItem?.route, currentItem?.navigationElements)
+
+                if (drawerContent == null) {
+                    Scaffold(
+                        topBar = topBar,
+                    ) {
+                        NavigationGraph(navController = navController)
+                    }
+                } else {
+                    NavigationDrawer(
+                        drawerContainerColor = Color.White,
+                        drawerTonalElevation = 0.dp,
+                        drawerShape = RectangleShape,
+                        drawerState = drawerState,
+                        drawerContent = drawerContent
+                    ) {
+                        Scaffold(
+                            topBar = topBar,
+                        ) {
+                            NavigationGraph(navController = navController)
+                        }
+                    }
+                }
```

Comment:
I'm afraid that with this approach, it's not possible to show other screens that aren't part of the "main" navigation tree like WelcomeScreen which doesn't show TopBar at all and LoginScreen which has a different TopBar than the MainScreen. 
I thought that every "fullscreen" screen should have its own TopBar since they are completely different and when from MainScreen we open a particular ConversationScreen it should replace the whole screen instead of just the content below the TopBar 🤔 
With this approach that I'm thinking about, we don't need to have single global navigation that handles all, we can have multiple nested ones, like: global app navigation can handle switching "fullscreen" screens (Splash, Login, Welcome, Main, Conversation, Call, Profile, Settings) and the MainScreen has its own inner navigation that handles only the content that should be shown inside the MainScreen - different items of the Drawer (Conversations, Vault, Archive), and then also ConversationsScreen can have another inner navigation that handles its tabs (All, Calls, Mentions). What do you think?